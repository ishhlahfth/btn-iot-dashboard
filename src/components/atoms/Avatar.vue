<template>
  <div
    v-if="!src"
    :style="avatarStyle"
    class="bg-midnight rounded-full flex flex-col justify-center items-center"
  >
    <p class="text-white text-subheading font-medium">{{ placeholder[0] }}</p>
  </div>
  <div v-else :style="avatarStyle" class="grid place-items-center">
    <img :src="src" alt="avatar" :style="avatarStyle" class="rounded-full object-cover w-full" />
  </div>
</template>

<script>
export default {
  name: 'HelpAvatar',
  props: {
    size: {
      type: Number,
      default: 64,
    },
    src: {
      type: String,
      default: '',
    },
    placeholder: {
      type: String,
      default: '',
    },
  },
  computed: {
    avatarStyle() {
      return {
        width: `${this.size.toString()}px`,
        height: `${this.size.toString()}px`,
      };
    },
  },
};
</script>

<style></style>
