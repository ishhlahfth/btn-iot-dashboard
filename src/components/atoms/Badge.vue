<template>
  <div
    class="flex justify-center max-w-min self-start rounded-full font-medium text-center text-xsmall py-1 px-2"
    :class="[
      { 'px-4': !icon },
      { 'text-mint-dark bg-mint-soft': color === 'positive' },
      { 'text-flame-dark bg-flame-soft': color === 'negative' },
    ]"
  >
    <div class="grid grid-flow-col auto-cols-max gap-1">
      <div v-if="icon" class="h-full w-full flex items-center justify-center">
        <icon :name="icon" />
      </div>
      <div class="h-full w-full flex items-center">
        <p class="capitalize">{{ label }}</p>
      </div>
    </div>
  </div>
</template>

<script>
import Icon from './Icon.vue';

export default {
  name: 'HelpBadge',
  components: {
    Icon,
  },
  props: {
    label: {
      type: String,
      default: '',
    },
    icon: {
      type: String,
      default: '',
    },
    color: {
      type: String,
      default: 'negative',
    },
  },
};
</script>
