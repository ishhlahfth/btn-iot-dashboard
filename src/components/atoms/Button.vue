<template>
  <button
    class="flex items-center justify-center font-semibold transition-all"
    :class="[
      `bg-${bgColor} text-${color}`,
      { bordered: outlined },
      iconOnly ? 'p-1 rounded-full' : 'py-2 px-4 rounded-lg',
      bgColor === 'transparent'
        ? 'hover:bg-grey-4 hover:bg-opacity-70'
        : 'hover:bg-opacity-80',
    ]"
  >
    <div class="grid grid-flow-col auto-cols-max gap-2">
      <div v-if="icon" class="h-full w-full flex items-center justify-center">
        <icon :name="icon" />
      </div>
      <div v-if="label" class="h-full w-full flex items-center">
        <p class="text-body">{{ buttonLabel }}</p>
      </div>
    </div>
  </button>
</template>

<script>
import Icon from './Icon.vue';

export default {
  name: 'HelpButton',
  props: {
    bgColor: {
      type: String,
      default: 'midnight',
    },
    color: {
      type: String,
      default: 'white',
    },
    outlined: {
      type: Boolean,
      default: false,
    },
    label: {
      type: String,
      default: '',
    },
    icon: {
      type: String,
      default: '',
    },
    iconOnly: {
      type: Boolean,
      default: false,
    },
  },
  components: {
    Icon,
  },
  computed: {
    buttonLabel() {
      return this.label.toUpperCase();
    },
  },
};
</script>

<style></style>
