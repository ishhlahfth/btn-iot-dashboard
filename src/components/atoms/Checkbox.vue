<template>
  <input
    @input="$emit('update:modelValue', !modelValue)"
    type="checkbox"
    :value="modelValue"
    class="help-checkbox appearance-none h-5 w-5 m-1 border text-royal border-grey-2 checked:bg-royal rounded focus:outline-none cursor-pointer"
  />
</template>

<script>
export default {
  name: 'HelpCheckbox',
  props: {
    modelValue: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style lang="scss" scoped>
.help-checkbox:checked {
  background-image: url(../../assets/icons/check.svg);
  border-color: transparent;
  background-size: 100% 100%;
  background-position: 50%;
  background-repeat: no-repeat;
}
</style>
