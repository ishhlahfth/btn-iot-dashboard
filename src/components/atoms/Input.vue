<template>
  <div class="grid grid-flow-row auto-rows-max gap-1">
    <label class="font-medium">{{ label }}</label>
    <div
      v-if="type === 'textarea'"
      class="border border-grey-4 py-2.5 px-3 rounded-lg grid grid-flow-col auto-cols-max gap-2 place-items-center"
      :class="[
        { 'ring-2 ring-royal ring-offset-1': onFocus },
        { 'ring-2 ring-flame ring-offset-1': hasError },
      ]"
      @click="$refs.helpInput.focus()"
      @blur="$refs.helpInput.blur()"
    >
      <textarea
        class="resize-none w-full"
        spellcheck="false"
        ref="helpInput"
        :placeholder="placeholder"
        :value="modelValue"
        :rows="rows"
        @blur="onFocus = false"
        @focus="onFocus = true"
        @input="$emit('update:modelValue', $event.target.value)"
      />
    </div>
    <div
      v-else
      class="border border-grey-4 py-2.5 px-3 rounded-lg grid grid-flow-col auto-cols-max gap-2 place-items-center"
      :class="[
        { 'ring-2 ring-royal ring-offset-1': onFocus },
        { 'ring-2 ring-flame ring-offset-1': hasError },
      ]"
      @click="$refs.helpInput.focus()"
      @blur="$refs.helpInput.blur()"
    >
      <slot name="prepend"></slot>
      <input
        ref="helpInput"
        spellcheck="false"
        :type="type"
        :placeholder="placeholder"
        :value="modelValue"
        @blur="onFocus = false"
        @focus="onFocus = true"
        @input="$emit('update:modelValue', $event.target.value)"
      />
      <slot name="append"></slot>
    </div>
    <p v-if="hasError" class="font-medium text-xsmall text-flame mt-0.5">
      This is an error message
      <!-- save this for later form validation -->
    </p>
  </div>
</template>

<script>
export default {
  name: 'HelpInput',
  props: {
    modelValue: {
      type: String,
      default: '',
    },
    label: {
      type: String,
      default: '',
    },
    placeholder: {
      type: String,
      default: '',
    },
    type: {
      type: String,
      default: 'text',
      required: true,
    },
    rows: {
      type: Number,
      default: 3,
    },
    // hasError: {
    //   type: Boolean,
    //   default: false,
    // }
  },
  data() {
    return {
      onFocus: false,
      hasError: false,
    };
  },
};
</script>

<style></style>
