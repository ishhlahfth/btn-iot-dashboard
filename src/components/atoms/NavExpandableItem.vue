<template>
  <div
    @click="collapsed = !collapsed"
    class="not-selected h-10 w-full rounded-md text-white p-2 flex items-center justify-between overflow-hidden cursor-pointer"
  >
    <div class="flex items-center whitespace-nowrap select-none">
      <div class="w-6 h-5 flex items-center justify-center mr-2">
        <icon name="photograph" :size="5" />
      </div>
      <div>
        <p class="font-medium">{{ menu.label }}</p>
      </div>
    </div>
    <div class="w-5 h-5 grid place-items-center">
      <icon name="chevron-down" :size="5" class="transition-all" :class="{ rotate: !collapsed }" />
    </div>
  </div>

  <template v-for="(subMenu, i) in menu.subMenu" :key="i">
    <router-link :to="subMenu.path" custom v-slot="{ href, navigate, isActive }">
      <div
        :active="isActive"
        :href="href"
        @click="navigate"
        v-if="!collapsed"
        class="nav-item h-10 w-full rounded-md text-white p-2 flex items-center justify-between overflow-hidden cursor-pointer"
        :class="{ 'bg-midnight-dark not-selectable': isActive }"
      >
        <div class="flex items-center whitespace-nowrap select-none">
          <div class="w-6 h-5 flex items-center justify-center mr-2"></div>
          <div>
            <p>{{ subMenu.label }}</p>
          </div>
        </div>
      </div>
    </router-link>
  </template>
</template>

<script>
import { RouterLink } from 'vue-router';
import Icon from './Icon.vue';

export default {
  name: 'HelpNavExpandableItem',
  components: {
    Icon,
    RouterLink,
  },
  props: {
    menu: {
      type: Object,
      required: true,
    },
    miniSidebar: {
      type: Boolean,
      default: false,
    },
  },
  data() {
    return {
      collapsed: true,
    };
  },
  watch: {
    miniSidebar() {
      if (this.miniSidebar) this.collapsed = true;
    },
  },
};
</script>

<style lang="scss" scoped>
.nav-item {
  &:hover {
    background: rgba($color: #003168, $alpha: .3);
  }
}
.not-selectable {
  &:hover {
    background: rgba($color: #003168, $alpha: 1);
  }
}
.slide-fade-enter-active,
.slide-fade-leave-active {
  transition: all 0.2s ease;
}
.slide-fade-enter-from,
.slide-fade-leave-to {
  transform: translateY(-10px);
  opacity: 0;
}
.rotate {
  transform: rotate(180deg);
}
</style>
