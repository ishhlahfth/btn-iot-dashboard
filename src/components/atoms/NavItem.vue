<template>
  <router-link :to="menu.path" custom v-slot="{ href, navigate, isActive }">
    <div
      :active="isActive"
      :href="href"
      @click="navigate"
      class="nav-item h-10 w-full rounded-md text-white p-2 flex items-center justify-between overflow-hidden cursor-pointer"
      :class="{ 'bg-midnight-dark not-selectable': isActive }"
    >
      <div class="flex items-center whitespace-nowrap select-none">
        <div class="w-6 h-5 flex items-center justify-center mr-2">
          <icon :name="menu.icon" :size="5" />
        </div>
        <div>
          <p class="font-medium">{{ menu.label }}</p>
        </div>
      </div>
    </div>
  </router-link>
</template>

<script>
import { RouterLink } from 'vue-router';
import Icon from './Icon.vue';

export default {
  name: 'HelpNavItem',
  components: {
    Icon,
    RouterLink,
  },
  props: {
    menu: {
      type: Object,
      required: true,
    },
  },
};
</script>

<style lang="scss" scoped>
.nav-item {
  &:hover {
    background: rgba($color: #003168, $alpha: .3);
  }
}
.not-selectable {
  &:hover {
    background: rgba($color: #003168, $alpha: 1);
  }
}
</style>
