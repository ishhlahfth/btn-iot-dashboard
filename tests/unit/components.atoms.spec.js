import { shallowMount } from '@vue/test-utils';
// testing components atoms
import Avatar from '@/components/atoms/Avatar.vue';
import Button from '@/components/atoms/Button.vue';
import Badge from '@/components/atoms/Badge.vue';
import Checkbox from '@/components/atoms/Checkbox.vue';
import Icon from '@/components/atoms/Icon.vue';
import Input from '@/components/atoms/Input.vue';
import Radio from '@/components/atoms/Radio.vue';
import Thumbnail from '@/components/atoms/Thumbnail.vue';
import Toggle from '@/components/atoms/Toggle.vue';
import Tooltip from '@/components/atoms/Tooltip.vue';
import OptionItem from '@/components/atoms/OptionItem.vue';
import NavItem from '@/components/atoms/NavItem.vue';
import NavExpandableItem from '@/components/atoms/NavExpandableItem.vue';

// testing props

describe('Button.vue', () => {
  it('should have "Button" as name', () => {
    expect(Button.name).toBe('HelpButton');
  });

  it('should render correctly', () => {
    const wrapper = shallowMount(Button);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render buttonLabel', () => {
    const wrapper = shallowMount(Button, {
      props: {
        label: 'hello world',
      },
    });

    expect(wrapper.vm.buttonLabel).toMatch('HELLO WORLD');
  });

  it('renders all props in button', () => {
    const wrapper = shallowMount(Button, {
      props: {
        bgColor: 'red-500',
        label: 'Search Address',
        color: 'midnight',
        icon: 'search',
        loading: true,
        disabled: false,
        type: 'submit',
      },
    });
    expect(wrapper.props().label).toBe('Search Address');
    expect(wrapper.props().color).toBe('midnight');
    expect(wrapper.props().disabled).toBe(false);
    expect(wrapper.props().loading).toBe(true);
  });
});

describe('Avatar.vue', () => {
  it('should have "Avatar" as name', () => {
    expect(Avatar.name).toBe('HelpAvatar');
  });

  it('should render correctly', () => {
    const wrapper = shallowMount(Avatar);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders all props in avatar', () => {
    const wrapper = shallowMount(Avatar, {
      props: {
        size: 80,
        src: 'https://images.pexels.com/photos/771742/pexels-photo-771742.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500',
        placeholder: 'Image Url',
      },
    });
    expect(wrapper.props().size).toBeGreaterThan(60);
    expect(wrapper.props().src).toMatch('images');
    expect(wrapper.props().placeholder).not.toMatch('Ulr');
  });
});

describe('Badge.vue', () => {
  it('should have "Badge" as name', () => {
    expect(Badge.name).toBe('HelpBadge');
  });

  it('should render correctly', () => {
    const wrapper = shallowMount(Badge);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders all props in badge', () => {
    const wrapper = shallowMount(Badge, {
      props: {
        label: 'chevron-up',
        icon: 'chevron-up',
        color: 'positive',
      },
    });
    expect(wrapper.props().label).toMatch('');
    expect(wrapper.props().icon).toMatch('');
    expect(wrapper.props().color).toBe('positive');
  });
});

describe('Checkbox.vue', () => {
  it('should have "Checkbox" as name', () => {
    expect(Checkbox.name).toBe('HelpCheckbox');
  });

  it('should render correctly', () => {
    const wrapper = shallowMount(Checkbox);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders all props in checkbox', () => {
    const wrapper = shallowMount(Checkbox, {
      props: {
        checked: true,
        label: '',
        disabled: false,
      },
    });
    expect(wrapper.props().checked).toBeTruthy();
    expect(wrapper.props().label).toMatch('');
    expect(wrapper.props().disabled).toBeFalsy();
  });
});

describe('Icon.vue', () => {
  it('should have "Icon" as name', () => {
    expect(Icon.name).toBe('HelpIcon');
  });

  it('should render correctly', () => {
    const wrapper = shallowMount(Icon);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders all props in icon', () => {
    const wrapper = shallowMount(Icon, {
      props: {
        name: 'search',
        size: 6,
      },
    });
    expect(wrapper.props().name).toMatch('');
    expect(wrapper.props().size).toBeGreaterThanOrEqual(4);
  });
});

describe('Input.vue', () => {
  it('should have "Input" as name', () => {
    expect(Input.name).toBe('HelpInput');
  });

  it('should render correctly', () => {
    expect(Input.$el).toMatchSnapshot();
  });

  // it('renders all props in input', () => {
  //   const wrapper = shallowMount(Input, {
  //     props: {
  //       modelValue: 'search',
  //       label: 'Search Address',
  //       placeholder: 'Jl.Bintara 14',
  //       type: 'text',
  //       rows: 3,
  //       leftIcon: 'search',
  //       rightIcon: '',
  //       searchBar: false,
  //       mask: '#####',
  //       background: 'white',
  //       disabled: true,
  //       pointer: false,
  //     },
  //   });
  //   expect(wrapper.props().modelValue).toMatch('');
  //   expect(wrapper.props().label).toMatch('');
  //   expect(wrapper.props().placeholder).toMatch('');
  //   expect(wrapper.props().type).toBe('text');
  //   expect(wrapper.props().rows).toBeGreaterThanOrEqual(3);
  //   expect(wrapper.props().leftIcon).toMatch('');
  //   expect(wrapper.props().rightIcon).toMatch('');
  //   expect(wrapper.props().searchBar).toBeFalsy();
  //   expect(wrapper.props().mask).toMatch('');
  //   expect(wrapper.props().background).toMatch('white');
  //   expect(wrapper.props().disabled).toBeTruthy();
  //   expect(wrapper.props().pointer).toBeFalsy();
  // });
});

describe('Radio.vue', () => {
  it('should have "Radio" as name', () => {
    expect(Radio.name).toBe('HelpRadio');
  });

  it('should render correctly', () => {
    const wrapper = shallowMount(Radio);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders all props in radio', () => {
    const wrapper = shallowMount(Radio, {
      props: {
        modelValue: false,
        group: '',
        label: '',
        disabled: false,
      },
    });
    expect(wrapper.props().modelValue).toBeFalsy();
    expect(wrapper.props().group).toMatch('');
    expect(wrapper.props().label).toMatch('');
    expect(wrapper.props().disabled).toBeFalsy();
  });
});

describe('Thumbnail.vue', () => {
  it('should have "Thumbnail" as name', () => {
    expect(Thumbnail.name).toBe('HelpThumbnail');
  });

  it('should render correctly', () => {
    const wrapper = shallowMount(Thumbnail);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders all props in thumbnail', () => {
    const wrapper = shallowMount(Thumbnail, {
      props: {
        width: 160,
        height: 90,
        src: 'https://images.pexels.com/photos/220453/pexels-photo-220453.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500',
        useBg: false,
      },
    });
    expect(wrapper.props().width).toBeGreaterThanOrEqual(100);
    expect(wrapper.props().height).toBeLessThanOrEqual(100);
    expect(wrapper.props().src).toMatch('');
    expect(wrapper.props().useBg).toBeFalsy();
  });
});

describe('Toggle.vue', () => {
  it('should have "Toggle" as name', () => {
    expect(Toggle.name).toBe('HelpToggle');
  });

  it('should render correctly', () => {
    const wrapper = shallowMount(Toggle);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders all props in toggle', () => {
    const wrapper = shallowMount(Toggle, {
      props: {
        modelValue: true,
      },
    });
    expect(wrapper.props().modelValue).not.toBeFalsy();
  });
});

describe('Tooltip.vue', () => {
  it('should have "Tooltip" as name', () => {
    expect(Tooltip.name).toBe('HelpTooltip');
  });

  it('should render correctly', () => {
    const wrapper = shallowMount(Tooltip);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders all props in tooltip', () => {
    const wrapper = shallowMount(Tooltip, {
      props: {
        text: 'Hello World',
      },
    });
    expect(wrapper.props().text).toMatch('');
  });
});

describe('OptionItem.vue', () => {
  it('should have "OptionItem" as name', () => {
    expect(OptionItem.name).toBe('HelpOptionItem');
  });

  it('should render correctly', () => {
    const wrapper = shallowMount(OptionItem);
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('NavItem.vue', () => {
  it('should have "NavItem" as name', () => {
    expect(NavItem.name).toBe('HelpNavItem');
  });

  it('should render correctly', () => {
    const wrapper = shallowMount(NavItem, {
      props: {
        menu: {
          path: '/',
          icon: 'search',
          label: 'dashboard',
        },
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('NavExpandableItem.vue', () => {
  it('should have "NavExpandableItem" as name', () => {
    expect(NavExpandableItem.name).toBe('HelpNavExpandableItem');
  });

  it('should render correctly', () => {
    expect(NavExpandableItem.$el).toMatchSnapshot();
  });

  it('renders all props in expandable item', async () => {
    const arrSubs = [
      {
        path: 'merchant',
        icon: 'search',
        label: 'merchant',
      },
    ];
    const wrapper = shallowMount(NavExpandableItem, {
      props: {
        menu: {
          path: '/',
          icon: 'search',
          label: 'dashboard',
          subMenu: arrSubs,
        },
        miniSidebar: false,
      },
    });
    expect(wrapper.props().miniSidebar).toBeFalsy();
  });
});
