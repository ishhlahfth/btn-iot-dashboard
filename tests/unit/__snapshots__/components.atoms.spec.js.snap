// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar.vue should render correctly 1`] = `
<div
  class="bg-midnight rounded-full flex flex-col justify-center items-center cursor-default select-none"
  style="width: 64px; height: 64px;"
>
  <p
    class="text-white text-subheading font-medium"
  />
  
  
</div>
`;

exports[`Badge.vue should render correctly 1`] = `
<div
  class="flex justify-center max-w-min self-start rounded-full font-medium text-center text-xsmall py-1 px-2 px-4 text-flame-dark bg-flame-soft"
>
  <div
    class="grid grid-flow-col auto-cols-max gap-1"
  >
    <!--v-if-->
    <div
      class="h-full w-full flex items-center"
    >
      <p
        class="capitalize"
      />
    </div>
  </div>
</div>
`;

exports[`Button.vue should render correctly 1`] = `
<button
  class="flex items-center justify-center font-semibold transition-all bg-midnight text-white py-2 px-4 rounded-lg hover:bg-opacity-80"
  type="submit"
>
  <div
    class="grid grid-flow-col auto-cols-max gap-2"
  >
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </div>
</button>
`;

exports[`Checkbox.vue should render correctly 1`] = `
<div
  class="grid grid-flow-col auto-cols-max gap-1 place-items-center"
>
  <input
    class="cursor-pointer bg-white help-checkbox appearance-none h-5 w-5 m-1 border text-royal border-grey-2 checked:bg-royal rounded focus:outline-none"
    type="checkbox"
  />
  <!--v-if-->
</div>
`;

exports[`Icon.vue should render correctly 1`] = `
<div
  class="w-4 h-4 border border-dashed rounded-sm"
/>
`;

exports[`Input.vue should render correctly 1`] = `undefined`;

exports[`NavExpandableItem.vue should render correctly 1`] = `undefined`;

exports[`NavItem.vue should render correctly 1`] = `
<router-link-stub
  ariacurrentvalue="page"
  custom="true"
  replace="false"
  to="/"
/>
`;

exports[`OptionItem.vue should render correctly 1`] = `
<p
  class="rounded py-2.5 px-4 w-full min-w-max cursor-pointer font-medium select-none hover:bg-grey-6"
/>
`;

exports[`Radio.vue should render correctly 1`] = `
<div
  class="grid grid-flow-col auto-cols-max gap-1 place-items-center"
>
  <input
    class="cursor-pointer bg-white help-radio appearance-none relative h-5 w-5 m-1 border text-royal border-grey-2 rounded-full checked:bg-royal checked:border-transparent focus:outline-none"
    name=""
    type="radio"
    value="false"
  />
  <!--v-if-->
</div>
`;

exports[`Thumbnail.vue should render correctly 1`] = `
<div
  class="border-2 border-grey-4 border-dashed rounded-md flex flex-col justify-center items-center"
  style="width: 160px; height: 90px; background-image: none; background-size: none;"
>
  
  <icon-stub
    name="photograph"
    size="4"
  />
  <p
    class="text-xsmall font-medium"
  >
    No Image
  </p>
  
</div>
`;

exports[`Toggle.vue should render correctly 1`] = `
<div
  class="w-11 p-0.5 flex rounded-full cursor-pointer transition-all duration-200 ease-in-out"
  style="background: rgb(228, 231, 235);"
>
  <input
    class="appearance-none cursor-pointer transition-all duration-200 ease-in-out"
    type="checkbox"
    value="false"
  />
  <label
    class="w-5 h-5 rounded-full bg-white shadow cursor-pointer"
  />
</div>
`;

exports[`Tooltip.vue should render correctly 1`] = `
<div
  class="relative flex tooltip-container"
>
  
  
  <div
    class="tooltip bg-grey-2 text-white text-center opacity-0 w-auto rounded py-1 px-3 absolute left-1/2 transform -translate-x-1/2 z-50 transition-opacity duration-75"
    style="bottom: 110%;"
  >
    <p
      class="text-xsmall"
    />
  </div>
</div>
`;
